src/Stack/Build/Execute.hs

```diff
@@ -881,21 +881,16 @@ ensureConfig newConfigCache pkgDir ExecuteEnv {..} announce cabal cabalfp task =
         announce
         cp <- view compilerPathsL
         let (GhcPkgExe pkgPath) = cpPkg cp
-        let programNames =
+        let exes =
               case cpWhich cp of
                 Ghc ->
                   [ "--with-ghc=" ++ toFilePath (cpCompiler cp)
                   , "--with-ghc-pkg=" ++ toFilePath pkgPath
                   ]
-        exes <- forM programNames $ \name -> do
-            mpath <- findExecutable name
```

Comment:
Here it'll run `findExecutable "--with-ghc=/path/to/ghc"`, which obviously will not return anything. Both filepaths are already absolute filepaths (on type level), so there's no reason to resolve anything here.

Reply:
I've made an alternative solution where I leave the `findExecutable` checking. I've just removed adding `--with-ghc/ghcpkg=` to the path, which obviously forces to fail the check: #5597

Reply:
I'm not sure if `findExecutable` on an absolute Path makes much sense (it probably just degrades to`doesFileExist` and checking the exec bit).

Moreover: if those paths don't point to valid ghc/ghci, you'd want proper failure, not hiding it by omitting the `--with-ghc` flag, no?

Reply:
> I'm not sure if `findExecutable` on an absolute Path

Does it have to be an absolute path, though?

Reply:
What do you mean? It **is** an absolute path always in this part of the code. That's ensured on type level.

Reply:
> I'm not sure if `findExecutable` on an absolute Path makes much sense (it probably just degrades to`doesFileExist` and checking the exec bit).

I think that checking the exec bit makes the error message more convenient for a user. 
What message will see the user if he runs a file without exec bit?

> Moreover: if those paths don't point to valid ghc/ghc-pkg, you'd want proper failure, not hiding it by omitting the `--with-ghc` flag, no?
> 
> One reason this bug was so hard to find is exactly the silent failure that causes other toolchain issues down the line.

I agree and in case of proper error messaging your patch is better.
In another case, I would throw an exception here.